Modules/Filtering/ImageIntensity/include/itkHistogramMatchingImageFilter.hxx

```diff
@@ -142,58 +116,108 @@ void
 HistogramMatchingImageFilter< TInputImage, TOutputImage, THistogramMeasurement >
 ::BeforeThreadedGenerateData()
 {
-  unsigned int j;
+  THistogramMeasurement sourceMeanValue;
+  THistogramMeasurement referenceMeanValue;
+
+  InputPixelType  sourceIntensityThreshold;
+  InputPixelType  referenceIntensityThreshold;
+
+  if ( m_GenerateReferenceHistogramFromImage )
```

Comment:
The best place to check if all the inputs are set at require is by overloading the [VerifyPreconditions ](https://itk.org/Doxygen/html/classitk_1_1ProcessObject.html#a73000f23064b7d2898fa21135080b4cf)method. This method gets called before any pipeline execution occurs even, `GenerateOutputInformation` and follows the fail quickly principle.